op-node/rollup/driver/sequencer.go

```diff
@@ -106,7 +106,7 @@ func (d *Sequencer) CreateNewBlock(ctx context.Context, l2Head eth.L2BlockRef, l
 	// TODO: allowing to breathe when remaining time is in the negative is very generous,
 	//  we can reduce this if the block building timing gets better with PR 3818
 	d.log.Debug("using remaining time for better block production", "remaining_time", remaining)
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(50 * time.Millisecond)
```

Comment:
It would be better if we adjusted the time based on whether or not we're close to the current actual wallclock time. If we are synced, we should take our sweet time with building the block. If we are not, then it makes sense to seal it ASAP, even if that means transactions might not have the chance to be included.

Reply:
Proposed algorithm for this:

- If we're > 1m away from wall clock time, no wait time - build immediately
- If we're < 1m away from wall clock time, 500ms as before

WDYT?

Reply:
Looked into it closer, and there is an edge case with low remaining time, and we should just cancel the job if the block time is further into the future than 2 block time spans.